### PR TITLE
Add support for .metrics in addition to .metric

### PR DESCRIPTION
**Pull request by yaron2**
Closes https://github.com/dapr/dapr/issues/5808.

### CODE REVIEW

**Comment by mukundansundar** [comment]
@yaron2 can you change existing e2e to use metrics instead of metric?

File: pkg/config/configuration.go [hunk 1]
```diff
@@ -98,6 +98,7 @@ type ConfigurationSpec struct {
 	TracingSpec         TracingSpec        `json:"tracing,omitempty" yaml:"tracing,omitempty"`
 	MTLSSpec            MTLSSpec           `json:"mtls,omitempty" yaml:"mtls,omitempty"`
 	MetricSpec          MetricSpec         `json:"metric,omitempty" yaml:"metric,omitempty"`
```
Comment (ItalyPaleAle) [inline pkg/config/configuration.go:100]: Should this be made a pointer so we can check if it's nil?

File: pkg/config/configuration.go [hunk 2]
```diff
@@ -336,9 +341,21 @@ func LoadKubernetesConfiguration(config, namespace string, podName string, opera
 		return nil, err
 	}
 
+	sortMetricsSpec(conf)
 	return conf, nil
 }
 
+// Apply .metrics if set. If not, retain .metric.
+func sortMetricsSpec(conf *Configuration) {
+	if !conf.Spec.MetricsSpec.Enabled {
```
Comment (ItalyPaleAle) [inline pkg/config/configuration.go:350]: I think that if we use .metrics for one thing, we should use it for all. With this code, a user could specify .metrics for one thing (for example, enabled), and .metric for the other (rules).

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]
Could you also add a warning log for people who are using .metric that they need to switch to .metrics?

**Comment by yaron2** [comment]
> @yaron2 can you change existing e2e to use metrics instead of metric?

I thought I did, with the testdata files. Can you point me to other locations?

File: pkg/config/configuration.go (hunk 2 above)
Comment (yaron2) [reply pkg/config/configuration.go:350]: The issue is that both metric and metrics are set to true by default, so we can't know if one field was set unless it's false, in which case all other settings are ignored. It's unlikely someone would use both, but if they do, it should be legal for as long as we support both I think.

File: pkg/config/configuration.go (hunk 1 above)
Comment (yaron2) [reply pkg/config/configuration.go:100]: In effect this can never be nil as we explicitly set it to true by default.

**Review by mukundansundar — APPROVED** [review]

**Review by artursouza — APPROVED** [review]

**Review by ItalyPaleAle — APPROVED** [review]

**Comment by mukundansundar** [comment]
@yaron2 I think a doc change is needed for this as well. 
cc @msfussell

**Comment by yaron2** [comment]
> @yaron2 I think a doc change is needed for this as well. cc @msfussell

Our docs have always mentioned `metrics`.

**Pull request merged**